dotnet/src/Connectors/Connectors.Memory.PgVector/PostgresServiceCollectionExtensions.cs

```diff
@@ -1,168 +1,320 @@
 ﻿// Copyright (c) Microsoft. All rights reserved.
 
+using System;
 using System.Diagnostics.CodeAnalysis;
 using Microsoft.Extensions.AI;
-using Microsoft.Extensions.DependencyInjection;
 using Microsoft.Extensions.VectorData;
+using Microsoft.SemanticKernel;
 using Microsoft.SemanticKernel.Connectors.PgVector;
 using Npgsql;
 
-namespace Microsoft.SemanticKernel;
+namespace Microsoft.Extensions.DependencyInjection;
 
 /// <summary>
 /// Extension methods to register Postgres <see cref="VectorStore"/> instances on an <see cref="IServiceCollection"/>.
 /// </summary>
 public static class PostgresServiceCollectionExtensions
 {
+    private const string DynamicCodeMessage = "This method is incompatible with NativeAOT, consult the documentation for adding collections in a way that's compatible with NativeAOT.";
+    private const string UnreferencedCodeMessage = "This method is incompatible with trimming, consult the documentation for adding collections in a way that's compatible with NativeAOT.";
+
     /// <summary>
-    /// Register a Postgres <see cref="VectorStore"/> with the specified service ID and where the NpgsqlDataSource is retrieved from the dependency injection container.
+    /// Register a <see cref="PostgresVectorStore"/> as <see cref="VectorStore"/>, where the <see cref="NpgsqlDataSource"/> is retrieved from the dependency injection container.
     /// </summary>
-    /// <param name="services">The <see cref="IServiceCollection"/> to register the <see cref="VectorStore"/> on.</param>
+    /// <param name="services">The <see cref="IServiceCollection"/> to register the <see cref="PostgresVectorStore"/> on.</param>
     /// <param name="options">Optional options to further configure the <see cref="VectorStore"/>.</param>
-    /// <param name="serviceId">An optional service id to use as the service key.</param>
+    /// <param name="lifetime">The service lifetime for the store. It needs to match <see cref="NpgsqlDataSource"/> lifetime. Defaults to <see cref="ServiceLifetime.Singleton"/>.</param>
```

Comment:
It's not quite correct that the lifetime here needss to match that of NpgsqlDataSource... It's perfectly OK to have Transient/Scoped here with a Singleton NpgsqlDataSource; and IIRC even if lifetimes are inversed (i.e. scoped NpgsqlDataSource resolved from Singleton store), DI "promotes" the NpgqlDataSource to Singleton by default (though there's an option to throw instead).

I'd basically remove this sentence... The summary already makes clear that this method resolves the NpgsqlDataSource from DI - for me that's sufficient.

```suggestion
    /// <param name="lifetime">The service lifetime for the store. Defaults to <see cref="ServiceLifetime.Singleton"/>.</param>
```